### PR TITLE
Whitespace check fix

### PR DESCRIPTION
**Pull request by abhinavagarwal07**
intentionally some whitespaces for testing purposes.
once the correctness is established, intentionally added whitespaces will be removed for final merge.
#830

### CODE REVIEW

**Comment by CodeMaxx** [comment]
This works but the reason whitespace check was added to the end of `test_travis.sh` was to get compiler errors and errors in Test before Travis exits due to whitespace errors. Can you try to get that working?
If not then this should be removed from Travis files and a python script,similar to one in #733 ,should be added instead.
@isuruf @certik Your thoughts on this?

**Comment by abhinavagarwal07** [comment]
There is a trade off. script is not working when used at end, but working well otherwise

**Comment by CodeMaxx** [comment]
Yeah I understand that but I believe we should rather add back the python script. It might lead to greater compile time in Travis,which is not a concern I think, but would serve both purposes.

**Comment by abhinavagarwal07** [comment]
the only reason python script was removed was that it was not working on travis.

File: bin/test_travis.sh [hunk 1]
```diff
@@ -100,13 +101,16 @@ ${CXX} -std=c++0x $compile_flags expand1.cpp $link_flags
 export LD_LIBRARY_PATH=$our_install_dir/lib:$LD_LIBRARY_PATH
 ./a.out
 
-echo "Checking whether all header files are installed:"
-python $SOURCE_DIR/symengine/utilities/tests/test_make_install.py $our_install_dir/include/symengine/ $SOURCE_DIR/symengine
-
+echo $SOURCE_DIR
+cd $SOURCE_DIR
```
Comment (Sumith1896) [inline bin/test_travis.sh]: Are these two lines needed?

File: bin/test_travis.sh [hunk 2]
```diff
@@ -100,13 +101,16 @@ ${CXX} -std=c++0x $compile_flags expand1.cpp $link_flags
...
 # check trailing whitespace:
-if !  egrep " $" -nr --include=\*.{cpp,h,inc}  --exclude-dir=*{teuchos,/build/}* $SOURCE_DIR ; then
+if !  egrep " $" -nr --include=\*.{cpp,h,inc}  --exclude-dir=*teuchos* --exclude-dir=*build* $SOURCE_DIR ; then
     echo No trailing whitespace;
 else
+    echo Please remove above stated whitespaces;
```
Comment (Sumith1896) [inline bin/test_travis.sh]: Why aren't the `echo` statements visible here https://travis-ci.org/symengine/symengine/jobs/112215038 ?

File: bin/test_travis.sh (hunk 1 above)
Comment (abhinavagarwal07) [inline bin/test_travis.sh]: since white-space check at _last_ is not working I'll have to revert the commit which added these lines anyways.

**Comment by abhinavagarwal07** [comment]
I have finally squashed commits and now though whitespace check will be at beginning but it will be working.

File: bin/test_travis.sh [hunk 3]
```diff
@@ -81,6 +81,7 @@ make
 echo "Running make install:"
 make install
 
+
```
Comment (CodeMaxx) [inline bin/test_travis.sh]: Can you remove the extra line to keep the spacings consistent.

**Comment by isuruf** [comment]
Can you figure out why it's not working when moved to the end?

Problem with having it at the beginning is that, sometimes we just need to see whether tests pass even if there are whitespace issues. Also it's easier to review PRs and to figure out an error from Travis run itself.

If we can't figure out a solution, next best option would be to have this check for just one job

**Comment by CodeMaxx** [comment]
Also remove the whitespaces currently present.
